crates/nu-cli/src/commands/classified/block.rs

```diff
@@ -40,30 +40,25 @@ pub async fn run_block(
                             inp,
                         )
                         .await?;
-                    loop {
-                        match output_stream.try_next().await {
-                            Ok(Some(ReturnSuccess::Value(Value {
-                                value: UntaggedValue::Error(e),
-                                ..
-                            }))) => return Err(e),
-                            Ok(Some(_item)) => {
-                                if let Some(err) = ctx.get_errors().get(0) {
-                                    ctx.clear_errors();
-                                    return Err(err.clone());
-                                }
-                                if ctx.ctrl_c.load(Ordering::SeqCst) {
-                                    break;
-                                }
+                    match output_stream.try_next().await {
+                        Ok(Some(ReturnSuccess::Value(Value {
+                            value: UntaggedValue::Error(e),
+                            ..
+                        }))) => return Err(e),
+                        Ok(Some(_item)) => {
+                            if let Some(err) = ctx.get_errors().get(0) {
+                                ctx.clear_errors();
+                                return Err(err.clone());
                             }
-                            Ok(None) => {
-                                if let Some(err) = ctx.get_errors().get(0) {
-                                    ctx.clear_errors();
-                                    return Err(err.clone());
-                                }
-                                break;
+                            if ctx.ctrl_c.load(Ordering::SeqCst) {}
```

Comment:
If we don't need the loop, we can also take this out as its only purpose here is to help us break the loop

Reply:
The issue is the existing code doesn't do anything with the Ctrl-C — so we can remove this and have the same results. 

But it should break out of the _outer_ loop. And if we add a `break` here, then we get some lifetime errors.

If my assertion is right (sounds like it might be given your comment above), I can look more at this — wanted to check given the chances of me vs the existing code making the mistake.

Reply:
I don't immediately see a test for this — where would be a good place to put one / are there any examples to base a test off?

Reply:
Unfortunately the best I could do is this, which would break but also return no output. Otherwise I'm spinning above my rust creds on lifetime issues:

```diff
diff --git a/crates/nu-cli/src/commands/classified/block.rs b/crates/nu-cli/src/commands/classified/block.rs
index f9f84abb..27901160 100644
--- a/crates/nu-cli/src/commands/classified/block.rs
+++ b/crates/nu-cli/src/commands/classified/block.rs
@@ -50,7 +50,9 @@ pub async fn run_block(
                                 ctx.clear_errors();
                                 return Err(err.clone());
                             }
-                            if ctx.ctrl_c.load(Ordering::SeqCst) {}
+                            if ctx.ctrl_c.load(Ordering::SeqCst) {
+                                return Ok(InputStream::empty());
+                            }
                         }
                         Ok(None) => {
                             if let Some(err) = ctx.get_errors().get(0) {
@@ -83,7 +85,9 @@ pub async fn run_block(
                                 ctx.clear_errors();
                                 return Err(err.clone());
                             }
-                            if ctx.ctrl_c.load(Ordering::SeqCst) {}
+                            if ctx.ctrl_c.load(Ordering::SeqCst) {
+                                return Ok(InputStream::empty());
+                            }
                         }
                         Ok(None) => {
                             if let Some(err) = ctx.get_errors().get(0) {

```

Let me know if you'd like me to apply this + a note, or we can merge this, or we can close the PR and I'll open an issue to track it.

Reply:
Yeah, an early return this way seems okay. Ctrl-C feels like the user wants to stop this thing as soon as possible.

Reply:
Great, done